commit db14e2664ed007251f721725bc810459a0f800c2	db14e2664ed007251f721725bc810459a0f800c2
Author: Gabor Buella <gabor.buella@oracle.com>
Date:   Thu Aug 24 10:02:48 2023 +0200

    Bug#35713721 redo log is never flushed
    
    (backport from trunk to 8.0)
    
    Flushing logs was skipped on Windows, since the value
    of Log_file_handle::s_skip_fsyncs was not in line
    with what is written in innobase/os/os0file.cc .
    
    On Windows, log files are always opened without
    the FILE_FLAG_NO_BUFFERING Windows attribute,
    thus they always need flushing, regardless what the
    value of SRV_WIN_IO_UNBUFFERED is.
    
    Change-Id: Ia8a608ccaec58c84453cffbeaa80842184d27c47

diff --git a/storage/innobase/log/log0log.cc b/storage/innobase/log/log0log.cc
index 279f46a6b13..8c951ba600f 100644
--- a/storage/innobase/log/log0log.cc
+++ b/storage/innobase/log/log0log.cc
@@ -1701,14 +1701,11 @@ dberr_t log_sys_init(bool expect_no_files, lsn_t flushed_lsn,
 #endif
       };
 
-#ifdef _WIN32
-  Log_file_handle::s_skip_fsyncs =
-      (srv_win_file_flush_method == SRV_WIN_IO_UNBUFFERED);
-#else
+#ifndef _WIN32
   Log_file_handle::s_skip_fsyncs =
       (srv_unix_file_flush_method == SRV_UNIX_O_DSYNC ||
        srv_unix_file_flush_method == SRV_UNIX_NOSYNC);
-#endif /* _WIN32 */
+#endif /* !_WIN32 */
 
   if (!found_files_in_root) {
     log_files_ctx =
